identity/src/main/java/com/android/identity/CredentialData.java

```diff
@@ -862,7 +902,89 @@ private boolean loadFromDisk(String dataKeyAlias) {
+            throw new RuntimeException("Error decoding ChunkedEncryptedData CBOR");
+        }
+        if (dataItems.size() != 1) {
+            throw new RuntimeException("Expected one item, found " + dataItems.size());
+        }
+        if (!(dataItems.get(0) instanceof Array)) {
+            throw new RuntimeException("Item is not a array");
+        }
+        Array array = (co.nstant.in.cbor.model.Array) dataItems.get(0);
+        if (array.getDataItems().size() < 2) {
+            throw new RuntimeException("Expected 2+ items, found " + array.getDataItems().size());
+        }
+        if (!(array.getDataItems().get(1) instanceof Array)) {
+            throw new RuntimeException("Second item in outer array is not a array");
+        }
+        Array innerArray = (co.nstant.in.cbor.model.Array) array.getDataItems().get(1);
+
+        try {
+            KeyStore ks = KeyStore.getInstance("AndroidKeyStore");
+            ks.load(null);
+            KeyStore.Entry entry = ks.getEntry(dataKeyAlias, null);
+            SecretKey secretKey = ((KeyStore.SecretKeyEntry) entry).getSecretKey();
+
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            for (DataItem item : ((Array) innerArray).getDataItems()) {
+                if (!(item instanceof ByteString)) {
+                    throw new RuntimeException("Item in inner array is not a bstr");
+                }
+                byte[] encryptedChunk = ((ByteString) item).getBytes();
+
+                ByteBuffer byteBuffer = ByteBuffer.wrap(encryptedChunk);
+                byte[] iv = new byte[12];
+                byteBuffer.get(iv);
+                byte[] cipherText = new byte[encryptedChunk.length - 12];
+                byteBuffer.get(cipherText);
+
+                Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+                cipher.init(Cipher.DECRYPT_MODE, secretKey, new GCMParameterSpec(128, iv));
+                byte[] decryptedChunk = cipher.doFinal(cipherText);
+                baos.write(decryptedChunk);
+            }
+            return baos.toByteArray();
+
+        } catch (InvalidAlgorithmParameterException
+                | NoSuchPaddingException
+                | BadPaddingException
+                | NoSuchAlgorithmException
+                | CertificateException
+                | InvalidKeyException
+                | IOException
+                | IllegalBlockSizeException
+                | UnrecoverableEntryException
+                | KeyStoreException e) {
+            throw new RuntimeException("Error decrypting chunk", e);
+        }
+    }
+
     private byte[] loadFromDiskDecrypt(String dataKeyAlias, byte[] encryptedFileData) {
+        // See saveToDiskEncrypt() for how newer versions of the library store the encrypted
+        // data in chunks.
+        final byte[] magic = CHUNKED_ENCRYPTED_MAGIC.getBytes(StandardCharsets.UTF_8);
+        if (encryptedFileData.length >= 2 + magic.length) {
```

Comment:
nit: it would be nice to have a (local) constant for `2` to make it "self documenting", just in case folks see it and wonder what the magic number is.

e.g. `final int CBOR_PREAMBLE_LENGTH = 1;` or `CBOR_BYTES_TO_SKIP`